deep_qa/layers/adaptive_recurrence.py

```diff
@@ -31,7 +31,8 @@ class AdaptiveRecurrence:
     can create one copy of the weights and although the number of iterations of the memory step is
     non-deterministic, the computational graph can be statically defined - we just loop over a part of it.
     '''
-    def __init__(self, memory_network, params: "Params"):
+    # pylint: disable=dangerous-default-value, unused-argument
```

Comment:
I think you don't need either of these pylint disablings - the first one is important, and we shouldn't disable it, and the second one shouldn't apply (unless it's for kwargs?).  I would do this instead like this:

```python
def __init__(self, memory_network, adaptive_step_params: Dict[str, Any]=None):
    if adaptive_step_params is None:
        adaptive_step_params = {}
    self.adaptive_step_params = adaptive_step_params
```

You could also consider just calling the argument `adaptive_step` instead of `adaptive_step_params`, letting the type convey the information that it's a set of parameters.